lib/ansible/modules/cloud/amazon/iam_cert.py

```diff
@@ -269,10 +281,23 @@ def main():
     cert_chain = module.params.get('cert_chain')
     dup_ok = module.params.get('dup_ok')
     if state == 'present':
-        cert = open(module.params.get('cert'), 'r').read().rstrip()
```

Comment:
Just noticed....   These assignments needs to be here (but modified) otherwise the following if statements will fail.
I would also move the cert_chain assignment into if state == 'present' block.

So along these lines....
```
    state = module.params.get('state')
    name = module.params.get('name')
    path = module.params.get('path')
    new_name = module.params.get('new_name')
    new_path = module.params.get('new_path')   
    dup_ok = module.params.get('dup_ok')

    if state == 'present':
        cert = module.params.get('cert')
        if cert is not None:
            if os.path.isfile(cert):
                cert = open(cert, 'r').read().rstrip()
            else:
                cert = to_text(cert.rstrip())

        key = module.params.get('key')
        if key is not None:
            if os.path.isfile(key):
                key = open(key, 'r').read().rstrip()
            else:
                key = to_text(key.rstrip())

        cert_chain = module.params.get('cert_chain')
        if cert_chain is not None:
            if os.path.isfile(cert_chain):
                cert_chain = open(cert_chain, 'r').read()
            else:
                cert_chain = to_text(cert_chain.rstrip())
    else:
        cert=key=cert_chain=None
```